android/app/src/main/java/com/reactnativenavigation/modal/RnnModal.java

```diff
@@ -87,4 +105,17 @@ public void onBackPressed() {
     public void onDismiss(DialogInterface dialog) {
         ModalController.getInstance().remove();
     }
+
+    @Override
+    public void dismiss(){
+        super.dismiss();
+        if (mContext != null && !mContext.isFinishing()) {
+            mContext.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    updateToolbar(mOriginalScreen, mOriginalToolbar);
```

Comment:
Ah. I was finding the visibility of the previous toolbar was being altered and I needed to reset it, but that might have been before I was setting the action toolbar context correctly on the modal. Let me have a play and see if I can clean this up.